### PR TITLE
create new access

### PR DESCRIPTION
**Pull request by tiagodavi**

### CODE REVIEW

File: nx/test/nx/tensor_test.exs [hunk 1]
```diff
@@ -127,4 +127,69 @@ defmodule Nx.TensorTest do
       assert_raise RuntimeError, "not supported", fn -> Nx.random_uniform({2, 2}) end
     end
   end
+
+  describe "access with :all" do
+    test "should access the entire tensor" do
+      t = Nx.tensor([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
+
+      assert t[[:all]] == t
```
Comment (polvalente) [inline nx/test/nx/tensor_test.exs:135]: @josevalim I'm not sure why this is behaving like this. Shouldn't this only bring the first row?

File: nx/test/nx/tensor_test.exs [hunk 2]
```diff
@@ -127,4 +127,69 @@ defmodule Nx.TensorTest do
...
+      assert t[[0..1, :all]] == Nx.tensor([[1, 2, 3], [4, 5, 6]])
+    end
+
+    test "should take the last two rows" do
+      t = Nx.tensor([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
+
+      assert t[[1..2, :all]] == Nx.tensor([[4, 5, 6], [7, 8, 9]])
+    end
+
+    test "should raise when out of bounds" do
```
Comment (polvalente) [inline nx/test/nx/tensor_test.exs:187]: I think we should have a higher dimensional test as well, like a 3D tensor having :all along each o the axes

File: nx/test/nx/tensor_test.exs (hunk 1 above)
Comment (jonatanklosko) [reply nx/test/nx/tensor_test.exs:135]: Isn't this conceptually the same as `t[:all]`, therefore taking all rows?

Comment (polvalente) [reply nx/test/nx/tensor_test.exs:135]: Oh right. So all implict dimensions are also ":all", right?

Comment (jonatanklosko) [reply nx/test/nx/tensor_test.exs:135]: Yup!

**Comment by josevalim** [comment]
Thank you @tiagodavi! Btw let's hold on a bit because we are likely going to add `..` to Elixir. So we can do:

    t[[.., 2]]

**Comment by josevalim** [comment]
Pull request in Elixir is here: https://github.com/elixir-lang/elixir/pull/11623

**Comment by josevalim** [comment]
Added to Elixir, thank you @tiagodavi for starting the discussion!

**Pull request closed without merging**